src/api/app/models/cloud/backend/upload_job.rb

```diff
@@ -32,8 +32,17 @@ def self.create(params)
         new(exception: exception.message)
       end
 
-      def self.all(user)
+      def self.find(job_id, options = {})
+        xml = ::Backend::Api::Cloud.upload_jobs([job_id])
+        return xml if options[:format] == :xml
+        new(xml_object: OpenStruct.new(Xmlhash.parse(xml)['clouduploadjob']))
```

Comment:
In case the job_id does not exist on the backend anymore, you return an empty UploadJob

Backend will return
```
"<clouduploadjoblist>\n</clouduploadjoblist>\n"
```

Which the Xmlhash turns into nil.

You would need to do sth like:

```
xml_hash = Xmlhash.parse(xml)['clouduploadjob'])
if hash.present? new(xml_object: OpenStruct.new(xml_hash)) : nil
```

I know that you check the existence of the job id in the frontend, but it could be that the data on the backend got deleted / destroyed...